src/compiler/compiler.rs

```diff
@@ -487,15 +474,26 @@ impl Compiler {
                     });
                 let future = Box::new(future);
                 Ok((CompileResult::CacheMiss(miss_type, duration, future), compiler_result))
+            }).chain_err(move || {
+                format!("failed to store `{}` to cache", out_file)
             }))
         }))
     }
 }
 
 /// Write `contents` to `path`.
```

Comment:
It'd be good to clarify here that this creates a temp dir and just writes a file under *that* path. This is a nice abstraction, though!